main.py

```diff
@@ -175,7 +176,62 @@ def execute():
+                        )))
+                    editions_addons_buttons = WebDriverWait(browser, TIMEOUT).until(
+                        EC.visibility_of_all_elements_located((
+                            By.XPATH,
+                            "//div[contains(@class,'Editions') or contains(@class, 'AddOns')]//div[contains(@class,'PurchaseButton-ctaButtons')]//button"
+                        )))
+                    if editions_addons_buttons[t].text == 'GET':
+                        editions_addons_buttons[t].click()
+
+                        # wait until its visible and then click the purchase button
+                        logger.debug('find and click on the last purchase button')
+                        WebDriverWait(browser, LOGIN_TIMEOUT).until(
+                            EC.visibility_of_element_located((By.XPATH, "//button[contains(@class,'btn-primary')]"))
+                        ).click()
+
+                        # wait until its visible and then click the 'I Agree" popup
+                        # 'Refund and Right of Withdrawal Information' popup
+                        try:
+                            logger.debug('accept the conditions of refund popup')
+                            WebDriverWait(browser, TIMEOUT).until(
+                                EC.visibility_of_all_elements_located(
+                                    (By.XPATH, "//button[contains(@class,'btn-primary')]"))
+                            )[1].click()
+                        except (NoSuchElementException, LookupError) as ex:
+                            logger.debug('no refund conditions popup to accept')
+
+                        # Purchase should be complete. Checking for confirmation
+                        logger.debug('Wait for confirmation that checkout is complete')
+
+                        # this is failing every time
+                        try:
+                            WebDriverWait(browser, LOGIN_TIMEOUT).until(EC.visibility_of_element_located((
+                                By.XPATH,
+                                "//h1/span[contains(text(),'Install')]|//span[contains(text(),'Thank you for buying')]"
+                            )))
+                        except (StaleElementReferenceException, TimeoutException) as e:
+                            # test if javascript refreshes the element in the meanwhile, which
+                            # requires another query
+                            try:
+                                WebDriverWait(browser, TIMEOUT).until(EC.visibility_of_element_located((
+                                    By.XPATH,
+                                    "//h1/span[contains(text(),'Install')]|//span[contains(text(),'Thank you for buying')]"
+                                )))
+                            except (StaleElementReferenceException, TimeoutException) as e:
+                                pass
```

Comment:
this block was copied over from somewhere else in the script thus could be made a function to reduce repeated code